.circleci/config.yml

```diff
@@ -37,6 +37,71 @@ workflows:
                 - linux-arm64
                 - mac
                 - windows
+  release-nightly:
+    triggers:
+      - schedule:
+          cron: "0 5 * * *"
+          filters:
+            branches:
+              only:
+                - main
+    jobs:
+      - lint
+      - check-openapi
+      - test:
+          matrix:
+            parameters:
+              executor:
+                - linux-amd64
+                - linux-arm64
+                - mac
+                - windows
+      - release:
+          is-snapshot: true
+          version: nightly-$(date +%Y-%m-%d)
+          requires:
+            - lint
+            - check-openapi
+            - test
+  release:
+    jobs:
+      - lint:
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^v[0-9]+\.[0-9]+\.[0-9]+(-(rc|alpha|beta)\.[0-9]+)?$/
+      - check-openapi:
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^v[0-9]+\.[0-9]+\.[0-9]+(-(rc|alpha|beta)\.[0-9]+)?$/
+      - test:
```

Comment:
It might be worth it to have `test` require `lint`? That way we aren't wasting cycles testing if linting fails?